core/services/workflows/engine_test.go

```diff
@@ -65,6 +65,44 @@ targets:
       abi: "receive(report bytes)"
 `
 
+// newTestEngine creates a new engine with some test defaults.
+func newTestEngine(t *testing.T, reg *coreCap.Registry, spec string) (eng *Engine, initFailed chan struct{}) {
+	peerID := p2ptypes.PeerID{}
+	initFailed = make(chan struct{})
```

Comment:
@cedric-cordenier would this be considered the proper way of converting a lifecycle hook to a signal? 

I could have used a signal handler directly rather than implementing a hook, but if we ever want to implement middleware to do things like o11y, then it'd be more complex to do so with channels rather than callbacks

Reply:
Yes lgtm! I agree that using functions internally makes sense :)